.github/workflows/publish.yaml

```diff
@@ -0,0 +1,56 @@
+# Creates and publishes assets for a new release.
+# To manually trigger this workflow, create a release in the GitHub UI.
+
+name: "Publish binaries"
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish-github:
+    strategy:
+      # Allowing jobs to fail until 'node-pre-gyp-github' supports failing gracefully if file already exists
+      # (https://github.com/bchr02/node-pre-gyp-github/issues/42)
+      fail-fast: false
+      matrix:
+        node_version: [12, 14, 16, 17]
+        experimental: [false]
+        system:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+        include:
+          - node_version: 16
+            system:
+              os: macos-latest
+              target: aarch64-apple-darwin
+            experimental: true
+    runs-on: ${{ matrix.system.os }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.release.tag_name }}
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{ matrix.node_version }}
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: ${{ matrix.system.target }}
+          override: true
+      - name: Install dependencies
+        working-directory: ./node-plugin
+        run: yarn install --ignore-scripts
+      - name: Compile binary, test, package, and publish to Github release page
```

Comment:
Node-pre-gyp-github doesn't fail gracefully if the file exists, so a lot of my CI tests have been failing.  Once a release pipeline is setup the publish workflow shouldn't need to be run multiple times, but I'd still like it to be idempotent.  I may explore using something like https://github.com/marketplace/actions/upload-files-to-a-github-release instead for uploading the binaries.  I think that update/exploration can happen in a subsequent PR though.